pkg/services/ngalert/store/loki_range_to_instant.go

```diff
@@ -20,20 +31,27 @@ func canBeInstant(r *models.AlertRule) bool {
 	if r.Data[0].QueryType != "range" {
 		return false
 	}
-	// First query part should go to cloud logs or insights.
+
+	var t dsType
+	// We can ignore the error here, the query just won't be optimized.
+	_ = json.Unmarshal(r.Data[0].Model, &t)
+
 	if r.Data[0].DatasourceUID != grafanaCloudLogs &&
 		r.Data[0].DatasourceUID != grafanaCloudUsageInsights &&
-		r.Data[0].DatasourceUID != grafanaCloudStateHistory {
-		return false
-	}
-	// Second query part should be and expression, '-100' is the legacy way to define it.
-	if r.Data[1].DatasourceUID != "__expr__" && r.Data[1].DatasourceUID != "-100" {
+		r.Data[0].DatasourceUID != grafanaCloudStateHistory &&
+		!t.isLoki() {
```

Comment:
Why keep all the UID checks? They're all already loki, can this be simplified to `if !t.isLoki() { return false }`?

Reply:
I thought about it but I was not sure if we always add the type, e.g. during migrations etc. But I think if we three all agree it should be fine. Will change it.